### PR TITLE
Disable radiative_model cache in NaimaSpectralModel init

### PR DESCRIPTION
**Pull request by QRemy**
Solves https://github.com/gammapy/gammapy/issues/3962

### CODE REVIEW

File: gammapy/modeling/models/spectral.py [hunk 1]
```diff
@@ -1979,11 +1967,17 @@ class NaimaSpectralModel(SpectralModel):
     tag = ["NaimaSpectralModel", "naima"]
 
     def __init__(
-        self, radiative_model, distance=1.0 * u.kpc, seed=None, nested_models=None
+        self,
+        radiative_model,
+        distance=1.0 * u.kpc,
+        seed=None,
+        nested_models=None,
+        use_cache=False,
```
Comment (adonath) [inline gammapy/modeling/models/spectral.py:1975]: Please add an entry in the docstring for this option...if you are confident this is not an option that user would need regularly, we could also just not expose it.

File: gammapy/modeling/models/tests/test_spectral.py [hunk 2]
```diff
@@ -53,7 +54,7 @@ def table_model():
 TEST_MODELS = [
     dict(
         name="constant",
-        model=ConstantSpectralModel(const=4 / u.cm**2 / u.s / u.TeV),
+        model=ConstantSpectralModel(const=4 / u.cm ** 2 / u.s / u.TeV),
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/modeling/models/tests/test_spectral.py]: yes but my version was outdated, modified

[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/modeling/models/tests/test_spectral.py]: Perfect, thanks for checking!

**Review by adonath — APPROVED** [review]
Thanks @QRemy, I have no further comments. Will merge after CI has passed...

**Pull request merged**